src/dotty/tools/dotc/core/OrderingConstraint.scala

```diff
@@ -438,8 +439,16 @@ class OrderingConstraint(private val boundsMap: ParamBounds,
     }
   }
 
-  def remove(pt: PolyType)(implicit ctx: Context): This =
-    newConstraint(boundsMap.remove(pt), lowerMap.remove(pt), upperMap.remove(pt))
+  def remove(pt: PolyType)(implicit ctx: Context): This = {
+    def removeFromOrdering(po: ParamOrdering) = {
+      def removeFromBoundss(key: PolyType, bndss: Array[List[PolyParam]]): Array[List[PolyParam]] = {
+        val bndss1 = bndss.map(bnds => bnds.filterConserve(_.binder ne pt))
+        if ((0 until bndss.length).forall(i => bndss(i) eq bndss1(i))) bndss else bndss1
+      }
+      po.remove(pt).mapValues(removeFromBoundss)
```

Comment:
I had to check the sources to find out, but `SimpleMap#mapValues` is eager, as opposed to `s.c.Map#mapValues`. That looks like what we want here.

I could imagine this subtle difference could get confusing when maintaining dotty. Might I suggest a different method name, like `mapValuesNow`? This might be the first call to this method, so a slightly longer name won't cause widespread ugliness.